src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -643,6 +643,8 @@ static void addERR_ParamsCantBeWithModifier(BindingDiagnosticBag diagnostics, Sy
             }
         }
 
+        /// <param name="typeWithAnnotations">Can be <see langword="default"/> if this is called for a lambda parameter
```

Comment:
> Can be

I think this part of the comment is misleading - it *will always be* default for an explicitly typed lambda parameter.

Consider also renaming to `typeWithAnnotationsOpt`.